app/components/channel_modal.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component, PropTypes} from 'react';
+import {Animated, Dimensions} from 'react-native';
+
+const AnimatedView = Animated.View;
+const {width: deviceWidth, height: deviceHeight} = Dimensions.get('window');
+
+export default class ChannelModel extends Component {
+    static propTypes = {
+        children: PropTypes.node,
+        duration: PropTypes.number,
+        topOffset: PropTypes.number,
+        theme: React.PropTypes.object.isRequired,
+        visible: PropTypes.bool.isRequired
+    }
+
+    static defaultProps = {
+        duration: 400,
+        topOffset: 0,
+        visible: false
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            top: new Animated.Value(deviceHeight)
+        };
+    }
+
+    componentWillReceiveProps(nextProps) {
+        if (this.props.visible === nextProps.visible) {
+            return;
+        }
+
+        const animateValue = nextProps.visible ? this.props.topOffset : deviceHeight;
+
+        Animated.timing(this.state.top, {
+            toValue: animateValue,
+            duration: this.props.duration
+        }).start();
+    }
+
+    render() {
+        return (
+            <AnimatedView style={{flex: 1, position: 'absolute', top: this.state.top, left: 0, height: (deviceHeight - this.props.topOffset), width: deviceWidth, backgroundColor: '#000', opacity: 0.95, overflow: 'hidden'}}>
```

Comment:
For this many styles, I would, at the very least, put each property on its own line. Especially because the values aren't purely static (we're accessing props/state and doing computation). Even though we're not enforcing an eslint rule for line length, to make our lives easier, I try to limit it to 80 chars whenever there are obvious, objective breakpoints (like in style objects & arrays).
For this component I would even opt for extracting the styles into an embedded stylesheet at the bottom. It's particularly appropriate for container components like this one.

@enahum, @hmhealey, @jarredwitt What preferences have you developed for formatting/organizing styles/stylesheets?